### PR TITLE
[JUJU-1666] Add a show-secret command

### PR DESCRIPTION
**Pull request by wallyworld**
Add a new show-secret command.
Modify the list-secret command to comply with the spec.
Also fix the handling of expire time updates.

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [X] Comments saying why design decisions were made
- [X] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- X ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps

```
juju deploy ubuntu

juju exec --unit ubuntu/0 "secret-add foo=bar"
secret:cbvfan6akragkndr0p30

juju exec --unit ubuntu/0 "secret-update secret:cbvfan6akragkndr0p30 foo=bar2"

 juju show-secret secret:cbvfan6akragkndr0p30 --revisions
cbvfan6akragkndr0p30:
  latest: 2
  owner: ubuntu
  create-time: 2022-08-19T02:28:45Z
  update-time: 2022-08-19T02:29:06Z
  revisions:
  - revision: 1
    create-time: 2022-08-19T02:28:45Z
    update-time: 2022-08-19T02:28:45Z
  - revision: 2
    create-time: 2022-08-19T02:29:06Z
    update-time: 2022-08-19T02:29:06Z

juju exec --unit ubuntu/0 "secret-update secret:cbvfan6akragkndr0p30 foo=bar3 --expire 1h"

juju show-secret secret:cbvfan6akragkndr0p30 --revisions
cbvfan6akragkndr0p30:
  latest: 3
  owner: ubuntu
  create-time: 2022-08-19T02:28:45Z
  update-time: 2022-08-19T02:29:30Z
  revisions:
  - revision: 1
    create-time: 2022-08-19T02:28:45Z
    update-time: 2022-08-19T02:28:45Z
  - revision: 2
    create-time: 2022-08-19T02:29:06Z
    update-time: 2022-08-19T02:29:06Z
  - revision: 3
    create-time: 2022-08-19T02:29:30Z
    update-time: 2022-08-19T02:29:30Z
    expire-time: 2022-08-19T03:29:30Z

juju show-secret secret:cbvfan6akragkndr0p30 --reveal
cbvfan6akragkndr0p30:
  latest: 3
  owner: ubuntu
  create-time: 2022-08-19T02:28:45Z
  update-time: 2022-08-19T02:29:30Z
  content:
    foo: bar3

juju show-secret secret:cbvfan6akragkndr0p30
cbvfan6akragkndr0p30:
  latest: 3
  owner: ubuntu
  create-time: 2022-08-19T02:28:45Z
  update-time: 2022-08-19T02:29:30Z

juju show-secret secret:cbvfan6akragkndr0p30 --revision 2 --reveal
cbvfan6akragkndr0p30:
  latest: 3
  owner: ubuntu
  create-time: 2022-08-19T02:28:45Z
  update-time: 2022-08-19T02:29:30Z
  content:
    foo: bar2

juju secrets 
ID                    Owner   Rotation  Latest  Last updated 
cbvfan6akragkndr0p30  ubuntu  never          3  4 minutes ago   
```

### CODE REVIEW

File: api/client/secrets/client.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/juju/juju/api/base"
 	"github.com/juju/juju/core/secrets"
 	"github.com/juju/juju/rpc/params"
+	secretsservice "github.com/juju/juju/secrets"
```
Comment (hpidcock) [inline api/client/secrets/client.go:12]: secretservice was a missed opportunity

File: secrets/provider/juju/secrets.go [hunk 2]
```diff
@@ -73,11 +73,38 @@ func (s secretsService) GetSecret(ctx context.Context, uri *coresecrets.URI) (*c
...
+		revs, err := s.backend.ListSecretRevisions(md.URI)
+		if err != nil {
+			return nil, nil, errors.Trace(err)
+		}
+		var filtered []*coresecrets.SecretRevisionMetadata
+		if filter.Revision == nil {
+			filtered = revs
+		} else {
+			for _, r := range revs {
+				if r.Revision != *filter.Revision {
```
Comment (hpidcock) [inline secrets/provider/juju/secrets.go:98]: would have been nice if this filter could have been passed into the ListSecretRevisions call

Comment (hpidcock) [reply secrets/provider/juju/secrets.go:98]: or a GetSecretRevision method

File: state/secrets.go [hunk 3]
```diff
@@ -195,10 +203,12 @@ func (s *secretsStore) secretRevisionDoc(uri *secrets.URI, revision int, expireT
 		DocID:      secretRevisionKey(uri, revision),
 		Revision:   revision,
 		CreateTime: s.st.nowToTheSecond(),
```
Comment (hpidcock) [inline state/secrets.go:205]: call nowToTheSecond once instead of twice

**Review by hpidcock — APPROVED** [review]

File: secrets/provider/juju/secrets.go (hunk 2 above)
Comment (wallyworld) [reply secrets/provider/juju/secrets.go]: added GetSecretRevision

**Pull request merged**